### PR TITLE
Use global transform for babelify

### PR DESCRIPTION
**Pull request by choumx**
Simplifies some code paths. 

https://github.com/ampproject/amphtml/pull/22848#discussion_r294011588

### CODE REVIEW

File: build-system/tasks/helpers.js [hunk 1]
```diff
@@ -404,6 +404,17 @@ function finishBundle(srcFilename, destDir, destFilename, options) {
   }
 }
 
+/**
+ * Returns array of relative paths to "dependencies" defined in package.json.
```
Comment (rsimha) [inline build-system/tasks/helpers.js]: nit: Should this comment say `"devDependencies"`?

Comment (choumx) [reply build-system/tasks/helpers.js]: Oops, thanks.

File: build-system/tasks/karma.conf.js [hunk 2]
```diff
@@ -68,7 +69,15 @@ module.exports = {
     debug: true,
     basedir: __dirname + '/../../',
     transform: [
-      ['babelify', {'global': isTravisBuild(), 'sourceMapsAbsolute': true}],
+      [
+        'babelify',
+        {
+          // Transform "node_modules/", but ignore devDependencies (on Travis).
+          'global': isTravisBuild(),
+          'ignore': devDependencies(),
```
Comment (choumx) [inline build-system/tasks/karma.conf.js]: I think this is a performance optimization.

File: build-system/tasks/karma.conf.js [hunk 3]
```diff
@@ -68,7 +69,14 @@ module.exports = {
     debug: true,
     basedir: __dirname + '/../../',
     transform: [
-      ['babelify', {'global': isTravisBuild(), 'sourceMapsAbsolute': true}],
+      [
+        'babelify',
+        {
+          'global': isTravisBuild(),
```
Comment (rsimha) [inline build-system/tasks/karma.conf.js]: Curious: Why Travis and not local? This is used for running tests in both cases.

Comment (choumx) [reply build-system/tasks/karma.conf.js]: Not sure, I actually didn't change this line.

Comment (rsimha) [reply build-system/tasks/karma.conf.js]: Yeah, what I meant was now that you're relying on this to transform `node_modules`, it's necessary to enable `global: true` for local testing and on Travis.

Comment (choumx) [reply build-system/tasks/karma.conf.js]: I think https://travis-ci.org/ampproject/amphtml/jobs/548325896 is failing because of this too.

Comment (rsimha) [reply build-system/tasks/karma.conf.js]: The link you posted confirms my suspicion. The failing runs are all coverage runs, which uses a different `babelify` config. I've sent out a fix in #22951, which I believe should handle this.

**Pull request merged**